### ISSUE TITLE
BlockGuardWithOnlyString should detect guards in non-DSL resources

### ISSUE THREAD

**Issue by tas50** [chef/cookstyle#190]
# Version:

5.1

# Environment:

macOS

# Scenario:

One would hope that everyone defines their resources using the DSL, but you can do it more programmatically using the classes directly like this:

```ruby
      def please_dont_do_this
        unless @gem_thing
          package_url = new_resource.package_url
          @gem_thing = Resource::ChefGem.new('a_gem', run_context)
          @gem_thing.version('~> 1.0')
          @gem_thing.only_if { 'a string' }
        end
        @gem_thing
      end
```

Our current matcher will not catch this. The matcher can be modified to allow .only_if to be called from another node like @gem_thing here, but the autocorrector needs some serious rework to make it work in this sort of scenario. See #189 for details.
